### PR TITLE
Veniq. Dataset validation. Fix return

### PR DESCRIPTION
**Pull request by lyriccoder**
When a function returns a value and the value is not assigned, it shouldn't add that line:
```
def inlined_func() {
                        if(_btSelectMethod == null) {
                                _btSelectMethod = new JButtonOperator(this, "...", 1);
                        }
                         return _btSelectMethod; // <- here
}
def original_func() {
                        func();
                        inlined_func();
                        func2();
}
```
Wrong:
```
def original_func() {
                        func();
                        if(_btSelectMethod == null) {
                                _btSelectMethod = new JButtonOperator(this, "...", 1);
                        }
                         _btSelectMethod; // <- here
                        func2();
}
```
Correct:
```
def original_func() {
                        func();
                        if(_btSelectMethod == null) {
                                _btSelectMethod = new JButtonOperator(this, "...", 1);
                        }
                        func2();
}
```

So, I fixed functionality and have written tests

### CODE REVIEW

**Comment by KatGarmash** [comment]
"When a function returns a value and the the value is not assigned, it shouldn't add that line:"
@lyriccoder i cant parse this sentence

File: test/dataset_collection/test_dataset_collection.py [hunk 1]
```diff
@@ -377,3 +361,69 @@ def test_start_end_inline_without_args(self):
...
+    def test_inline_with_return_without_assigning(self):
+        filepath = self.current_directory / 'InlineExamples' / 'Parameters.java'
+        test_filepath = self.current_directory / 'InlineTestExamples' / 'Parameters_without_return.java'
+        algorithm_type = InlineTypesAlgorithms.WITH_RETURN_WITHOUT_ARGUMENTS
+        algorithm_for_inlining = AlgorithmFactory().create_obj(algorithm_type)
+        ast = AST.build_from_javalang(build_ast(filepath))
+        m_decl = [
+            x for x in ast.get_proxy_nodes(ASTNodeType.METHOD_DECLARATION)
+            if x.name == 'btSelectMethod'][0]
+        body_start_line, body_end_line = method_body_lines(m_decl, filepath)
```
Comment (aravij) [inline test/dataset_collection/test_dataset_collection.py:374]: Here are some code duplication among your test function. Might be better to extract.

File: test/dataset_collection/test_dataset_collection.py [hunk 2]
```diff
@@ -377,3 +361,69 @@ def test_start_end_inline_without_args(self):
...
+        self.assertEqual(body_start_line == 80, body_end_line == 85)
+        algorithm_for_inlining().inline_function(
+            filepath,
+            111,
```
Comment (aravij) [inline test/dataset_collection/test_dataset_collection.py:378]: What a magic number?

**Review by aravij — COMMENTED** [review]
Just few suggestions.

File: test/dataset_collection/test_dataset_collection.py (hunk 2 above)
Comment (lyriccoder) [reply test/dataset_collection/test_dataset_collection.py:378]: invocation_line

File: test/dataset_collection/test_dataset_collection.py (hunk 1 above)
Comment (lyriccoder) [reply test/dataset_collection/test_dataset_collection.py:374]: unfortunately, we have side effects here. Also, there are at least 3 variables are used after invocation.
 Seems it's a bad idea. Moreover, it's a test but not functionality of some framework.
 Finally, if I extract that functionality, it doesn't follow single responsibility rule

Comment (lyriccoder) [reply test/dataset_collection/test_dataset_collection.py]: I tried to extract some code. But I do not know whether it's a good idea

**Review by aravij — APPROVED** [review]

**Pull request merged**